action/protocol/staking/contractstaking/cache.go

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2023 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package contractstaking
+
+import (
+	"math/big"
+
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/pkg/errors"
+)
+
+type (
+	contractStakingCache struct {
+		idBucketMap           map[uint64]*bucketInfo      // map[token]bucketInfo
+		candidateBucketMap    map[string]map[uint64]bool  // map[candidate]bucket
+		idBucketTypeMap       map[uint64]*BucketType      // map[token]BucketType
```

Comment:
just `bucketTypeMap`? the `id` here is NOT the same as `id` in L17, right? If yes, then the naming could be a little confusing